server/task_executor_schema_update_ghost_sync.go

```diff
@@ -211,7 +211,7 @@ func getGhostConfig(task *api.Task, dataSource *api.DataSource, userList []*api.
 
 func (exec *SchemaUpdateGhostSyncTaskExecutor) runGhostMigration(ctx context.Context, server *Server, task *api.Task, statement string) (terminated bool, result *api.TaskRunResultPayload, err error) {
 	syncDone := make(chan struct{})
-	migrationError := make(chan error)
+	migrationError := make(chan error, 1)
```

Comment:
Why is this change needed?

Reply:
If the task is canceled, no one will listen to `migrationError` channel, which blocks the sender @L273-283.

Another way is to send it if not canceled. Then this change is not needed.

Reply:
Can we cancel the gh-ost migrator with context.Context?

Reply:
If by migrator you mean `migrator.Migrate()`, I think we can't do that.

Reply:
SG. Please add a comment here for the change.